zookeeper-server/src/main/java/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -274,7 +274,7 @@ public void respondToServer(byte[] serverToken, ClientCnxn cnxn) {
             } catch (SaslException e) {
                 LOG.error("SASL authentication failed using login context '"
                           + this.getLoginContext()
-                          + "' with exception: {}", e);
```

Comment:
I think slf4 still logs the exception as exceptions are handled in a special way.
Doesn't it?

Reply:
![image](https://user-images.githubusercontent.com/7615397/64586920-4a04d600-d3d0-11e9-87be-1f880bc6c8c6.png)
Just like this.

Reply:
@eolivelli there are several overload methods, what are relevant is as below

```java
public void error(String format, Object... arguments);
public void error(String msg, Throwable t);
```

I think

```java
LOG.error(String.format("... %s", e.getMessage()), e);
```

is also a valid fix.

Reply:
Ya, this example is a bit odd.

```
                LOG.error("SASL authentication failed using login context '"
                          + this.getLoginContext()
                          + "' with exception: {}", e);
```

It is using parameters and concatenation.  Typically one or the other is used, not both.

`Throwable` objects are handled special.  They do not need the marker `{}` to be included in the message, they just need to be the last argument in the list.  If a stack trace is not desirable, then  using a marker in conjunction with `e.getMessage()` can be used.

Something better would be:
```
LOG.error("SASL authentication failed using login context '{}'", this.getLoginContext(), e);
```

The login context is passed to the marker, and the `Exception` message (plus stack trace) is printed as well.